pkg/apis/pipeline/v1beta1/task_conversion.go

```diff
@@ -34,14 +41,17 @@ func (t *Task) ConvertTo(ctx context.Context, to apis.Convertible) error {
 	switch sink := to.(type) {
 	case *v1.Task:
 		sink.ObjectMeta = t.ObjectMeta
-		return t.Spec.ConvertTo(ctx, &sink.Spec)
+		return t.Spec.ConvertTo(ctx, &sink.Spec, &sink.ObjectMeta, t.Name)
 	default:
 		return fmt.Errorf("unknown version, got: %T", sink)
 	}
 }
 
 // ConvertTo implements apis.Convertible
-func (ts *TaskSpec) ConvertTo(ctx context.Context, sink *v1.TaskSpec) error {
+func (ts *TaskSpec) ConvertTo(ctx context.Context, sink *v1.TaskSpec, meta *metav1.ObjectMeta, taskName string) error {
+	if err := serializeTaskDeprecations(meta, ts, taskName); err != nil {
```

Comment:
shall we serialize in #line 42 instead of in the ConvertTo as the other serilizations do? just suggesting since this could be up to your style choice

Reply:
This wouldn't work because we also need to deal with those deprecated fields for inline taskSpec definitions like `taskrun.spec.taskspec`, or `pipeline.spec.taskspec` etc.